apps/devhub/views.py

```diff
@@ -1758,14 +1759,16 @@ def api_key(request):
     if request.user.read_dev_agreement is None:
         return redirect(reverse('devhub.api_key_agreement'))
 
-    user_id = None
-    secret = None
-
     if request.method == 'POST':
-        # TODO: Generate an API key
-        pass
+        # TODO: support delete/regenerate. bug 1209224.
+        APIKey.new_jwt_credentials(request.user)
+        return redirect(reverse('devhub.api_key'))
```

Comment:
It's a POST, of course it should redirect. Don't mind me or my stupid comments ;)